### PR TITLE
[Catalog] Avoid cropping on the ShadowLayer example

### PR DESCRIPTION
**Pull request by fluxminds**
cropped on small device; fix the topLayout issue (the top label covered by the app bar)

closes (https://github.com/material-components/material-components-ios/issues/386)

Before the change:
simulator screen shot - iphone 5 - 2018-08-24 at 10 18 58
simulator screen shot - iphone 5 - 2018-08-24 at 10 18 53

After the change:
simulator screen shot - iphone 5 - 2018-08-24 at 10 20 18
simulator screen shot - iphone 5 - 2018-08-24 at 10 20 22

### CODE REVIEW

File: components/ShadowElevations/examples/ShadowElevationsTypicalUseExample.m [hunk 1]
```diff
@@ -124,10 +201,65 @@ - (void)viewDidLoad {
...
+#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
+- (void)viewSafeAreaInsetsDidChange {
+  [super viewSafeAreaInsetsDidChange];
+  CGRect insetedShadowViewFrame = CGRectMake(self.view.bounds.origin.x,
+                                             self.view.bounds.origin.y + self.view.safeAreaInsets.top,
+                                             self.view.bounds.size.width - self.view.safeAreaInsets.left - self.view.safeAreaInsets.right,
+                                             self.view.bounds.size.height - self.view.safeAreaInsets.top - self.view.safeAreaInsets.bottom);
+  self.shadowsView.frame = insetedShadowViewFrame;
 }
+#endif
```
Comment (fluxminds) [inline components/ShadowElevations/examples/ShadowElevationsTypicalUseExample.m]: Not sure if there is a better method I can utilize to layout with safeArea in the view update life cycle here.

**Comment by jverkoey** [comment]
Before + After screenshots please.

@romoore  used a new table format that was pretty nice:

```
| Before | After |
|:-------|:------|
| <before_screenshot> | <after_screenshot> |
```

Preview:

| Before | After |
|:-------|:------|
| <before_screenshot> | <after_screenshot> |

**Comment by romoore** [comment]
Please clean-up the summary and commit message. 

Helpful article: https://chris.beams.io/posts/git-commit/

**Comment by fluxminds** [comment]
@romoore Thanks for sharing. Good readings! I will use a more precise commit message when squashing the commits on this branch.

File: components/ShadowElevations/examples/ShadowElevationsTypicalUseExample.m [hunk 2]
```diff
@@ -124,10 +201,65 @@ - (void)viewDidLoad {
   self.view.backgroundColor = [UIColor whiteColor];
   self.title = @"Shadow Elevations";
   _shadowsView = [[ShadowElevationsPointsView alloc] initWithFrame:self.view.bounds];
-  _shadowsView.autoresizingMask =
-      UIViewAutoresizingFlexibleHeight | UIViewAutoresizingFlexibleWidth;
   [self.view addSubview:_shadowsView];
+  
+#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
```
Comment (codeman7) [inline components/ShadowElevations/examples/ShadowElevationsTypicalUseExample.m]: I don't think you need any of these `#if` checks or the `if (@available(iOS 11.0, *))` check.

Comment (fluxminds) [reply components/ShadowElevations/examples/ShadowElevationsTypicalUseExample.m]: Thanks for the comment! I feel this part is confusing too. What I wanted to do was using the `autoresizingMask` when it is `>= iOS11` and let the `viewSafeAreaInsetsDidChange` to handle `shadowView`'s position. When the OS is below 11, it uses `setupShadowsViewConstraints` to set up the constraint. How do you feel about it?

Comment (fluxminds) [reply components/ShadowElevations/examples/ShadowElevationsTypicalUseExample.m]: I just realized `#if ` is not needed anymore. I misunderstood it, and please ignore my previous comment. Thanks for pointing it out!

Comment (ianegordon) [reply components/ShadowElevations/examples/ShadowElevationsTypicalUseExample.m]: Please remove #if defined(__IPHONE_11_0).... guards.
We no longer need them as we no longer support Xcode 8.

**Review by ianegordon — APPROVED** [review]
Please address comment before merging.

**Comment by fluxminds** [comment]
Comments have been addressed in https://github.com/material-components/material-components-ios/pull/4884/commits/58543178dc33b1d1bfbcf5dcc483a32a52be03a1

**Pull request merged**